Microsoft.DotNet.Interactive/KernelExtensionLoader.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using MLS.Agent.Tools;
+using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.Loader;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Interactive
+{
+    public class KernelExtensionLoader
+    {
+        public async Task<bool> TryLoadFromAssembly(FileInfo assemblyFile, IKernel kernel)
+        {
+            if (assemblyFile == null)
+            {
+                throw new ArgumentNullException(nameof(assemblyFile));
+            }
+
+            if (kernel == null)
+            {
+                throw new ArgumentNullException(nameof(kernel));
+            }
+
+            var assembly = AssemblyLoadContext.Default.LoadFromAssemblyPath(assemblyFile.FullName);
+            var extensionTypes = assembly
+                                   .ExportedTypes
+                                   .Where(t => typeof(IKernelExtension).IsAssignableFrom(t))
+                                   .ToArray();
+
+            foreach (var extensionType in extensionTypes)
+            {
+                var extension = (IKernelExtension)Activator.CreateInstance(extensionType);
+                await extension.OnLoadAsync(kernel);
+            }
+
+            if (extensionTypes.Count() > 0)
```

Comment:
It's better to use `Length` than `Count()` here. The latter is a LINQ method and triggers iteration, while the former is a property of the array.

You can also simplify the branching logic to:

```csharp
return extensionTypes.Length > 0;
```

Reply:
Done. Thanks